runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java

```diff
@@ -91,6 +99,18 @@ public NotificationDto process(File runDirectory, TimeZone tz) throws IOExceptio
         .flatMap(parameters -> Arrays.stream(IlluminaChemistry.values()).filter(chemistry -> chemistry.test(parameters)).findFirst())
         .orElse(IlluminaChemistry.UNKNOWN));
 
+    // See if we can figure out a sample sheet
+    dto.setPoolNames(Optional.of(new File(runDirectory, "SampleSheet.csv"))//
+        .filter(File::canRead)//
+        .map(File::toPath)
+        .map(WhineyFunction.log(log, Files::lines))//
+        .orElse(Stream.empty())//
+        .filter(LimsUtils.rejectUntil(line -> line.startsWith("Sample_ID,")))//
+        .map(COMMA::split)//
+        .map(Pair.number(1))
+        .filter(pair -> pair.getValue().length > 0)//
+        .collect(Collectors.toMap(Entry::getKey, e -> e.getValue()[0])));
+
     // The Illumina library can't distinguish between a failed run and one that either finished or is still going. Scan the logs, if
     // available to determine if the run failed.
     File rtaLogDir = new File(runDirectory, "/Data/RTALogs");
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Refactor your code to get this URI from a customizable parameter. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1075)